src/etl/Source.h

```diff
@@ -339,13 +337,17 @@ class SourceImpl : public Source
             if (ec)
                 return {};
 
-            // Set a decorator to change the User-Agent of the handshake and to tell rippled to charge the client IP for
-            // RPC resources. See "secure_gateway" in
+            // if client ip is know, change the User-Agent of the handshake and to tell rippled to charge the client IP
+            // for RPC resources. See "secure_gateway" in
             // https://github.com/ripple/rippled/blob/develop/cfg/rippled-example.cfg
-            ws->set_option(websocket::stream_base::decorator([&clientIp](websocket::request_type& req) {
-                req.set(http::field::user_agent, std::string(BOOST_BEAST_VERSION_STRING) + " websocket-client-coro");
-                req.set(http::field::forwarded, "for=" + clientIp);
-            }));
+            if (clientIp)
+            {
+                ws->set_option(websocket::stream_base::decorator([&clientIp](websocket::request_type& req) {
+                    req.set(
+                        http::field::user_agent, std::string(BOOST_BEAST_VERSION_STRING) + " websocket-client-coro");
```

Comment:
This parameter is to notify the request's real owner. Rippled has the protection to avoid overwhelm the server. When client ip is not known, it is to say the request is from clio server itself. Usually clio server is in the secure_gateway cfg.